tools/configuration.py

```diff
@@ -1,24 +1,40 @@
 import os
 import sys
-import getopt
+import argparse
 
 from tools import get_pipe_output
 
-default_conf = {
-    'max_domains': 10,
-    'max_ext_length': 10,
-    'style': 'gitstats.css',
-    'max_authors': 7,
-    'max_authors_of_months': 6,
-    'authors_top': 5,
-    'commit_begin': '',
-    'commit_end': 'HEAD',
-    'linear_linestats': 1,
-    'project_name': '',
-    'processes': 8,
-    'start_date': '',
-    'output': 'html'
-}
+GITSTAT_VERSION = '0.91'
+
+
+def get_gitstat_parser() -> argparse.ArgumentParser:
```

Comment:
You have added all possible options (even unused ones) to the arguments. Let us make it simpler and give user a possibility only to configure important thigs from command line, e.g.

- path to repo
- output path for generated report and data
- version
- maybe path to config file in json-format with all the rest of options

Reply:
Could I implement this in other PR? Other change request are fixed now.

Reply:
I add config file feature too in this PR. The solution is come from your recommendation for directory input param validation below :)  It was really useful.